topdown/crypto.go

```diff
@@ -378,43 +416,57 @@ func getX509CertsFromPem(pemBlocks []byte) ([]*x509.Certificate, error) {
 	return x509.ParseCertificates(decodedCerts)
 }
 
-func getRSAPrivateKeyFromString(key string) (interface{}, error) {
-	// if the input is PEM handle that
+func getPrivateKeyFromString(key string) (interface{}, error) {
+	lst := [][]byte{}
```

Comment:
As discussed here: https://github.com/open-policy-agent/opa/pull/6084#discussion_r1260959636 I think that it'd be best to use the PEM block parsing loop I recommended. 

At the moment, the implementation adds all the input to a `[][]byte{}`, and then passes this `getPrivateKeyFromPem` which looks at the first block and discards the rest.

This is comparable to the implementation of `crypto.x509.parse_certificates` which is based on `x509.ParseCertificates`.

For example: https://play.openpolicyagent.org/p/QGkoAZb5RH